File: market/test/java/ru/yandex/market/hrms/core/service/environment/EnvironmentServiceTest.java
package ru.yandex.market.hrms.core.service.environment;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;
import org.springframework.beans.factory.annotation.Autowired;

import ru.yandex.market.common.test.db.DbUnitDataSet;
import ru.yandex.market.hrms.core.AbstractCoreTest;
import ru.yandex.market.hrms.core.service.environment.repo.EnvironmentKey;

class EnvironmentServiceTest extends AbstractCoreTest {

    @Autowired
    private EnvironmentService environmentService;

    @DbUnitDataSet(before = "EnviromentServiceTest.before.csv")
    @CsvSource(value = {
            "CALENDAR_FOR_NEXT_MONTH,,true",
            "CALENDAR_FOR_NEXT_MONTH,2,true",
            "CALENDAR_FOR_NEXT_MONTH,1,false",
            "CALENDAR_FOR_NEXT_MONTH,3,false"
    })
    @ParameterizedTest
    void getBooleanOrDefault(EnvironmentKey environmentKey, Long domainId, Boolean expected) {
        Boolean result = environmentService.getBooleanOpt(environmentKey, domainId).orElse(null);
        Assertions.assertEquals(expected, result);
    }

}

File: market/test/kotlin/ru/yandex/market/ext/marketplace/integrator/app/environment/EnvironmentServiceTest.kt
package ru.yandex.market.ext.marketplace.integrator.app.environment

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import ru.yandex.market.common.test.db.DbUnitDataSet
import ru.yandex.market.ext.marketplace.integrator.app.AbstractFunctionalTest

@DbUnitDataSet(before = ["environment-service-test-before.csv"])
class EnvironmentServiceTest : AbstractFunctionalTest() {

    @Autowired
    lateinit var environmentService: EnvironmentService

    @Test
    fun testGetBoolean() {
        Assertions.assertThat(environmentService.getBoolean("boolKey1")).isTrue
        Assertions.assertThat(environmentService.getBoolean("boolKey2")).isTrue
        Assertions.assertThat(environmentService.getBoolean("boolKey3")).isTrue
        Assertions.assertThat(environmentService.getBoolean("boolKey4")).isFalse
    }

    @Test
    fun testGetLong() {
        Assertions.assertThat(environmentService.getLong("longKey1")).isEqualTo(123456L)
        Assertions.assertThatExceptionOfType(NumberFormatException::class.java).isThrownBy {
            environmentService.getLong("longKey2")
        }
    }

    @Test
    fun testSetValue() {
        Assertions.assertThat(environmentService.getString("testKey1")).isEqualTo("value1")
        environmentService.set("testKey1", "value2")
        Assertions.assertThat(environmentService.getString("testKey1")).isEqualTo("value2")
    }
}
